rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -208,14 +212,56 @@ public void setAllCached(String remotePath, long fileLength, long lastModified,
     public Map getCacheStats()
     {
         Map<String, Double> stats = new HashMap<String, Double>();
-        stats.put("Cache Hit Rate", ((double) cachedRequests / totalRequests));
-        stats.put("Cache Miss Rate", ((double) (totalRequests - cachedRequests) / totalRequests));
+        stats.put("Cache Hit Rate", ((double) cachedRequests / (cachedRequests + remoteRequests)));
+        stats.put("Cache Miss Rate", ((double) (remoteRequests) / (cachedRequests + remoteRequests)));
         stats.put("Cache Reads", ((double) cachedRequests));
         stats.put("Remote Reads", ((double) remoteRequests));
         stats.put("Non-Local Reads", ((double) (totalRequests - cachedRequests - remoteRequests)));
         return stats;
     }
 
+    //This method is to ensure that data required by another node is cached before it is read by that node
+    //using localTransferServer.
+    // If the data is not already cached, remoteReadRequest for that block is sent and data is cached.
+
+    //TODO : buffer is initialized everytime readData is called and it contains garbage value which is not required. Also, getCacheStatus is called twice.
```

Comment:
explain a bit more about why we cannot mark it as static. Add enough context so that when a new person sees this it is easy to pick up the work